src/world/GameWorldView.cpp

```diff
@@ -267,67 +275,71 @@ void GameWorldView::DrawGUI(const RoadBuildState& rb, const TerrainRenderer& ter
                         default: break;
                     }
                 }
-
                 LOADER.GetMapImageN(mid)->DrawFull(curPos);
             }
 
             // Currently selected point
             if(selectedPt == curPt)
                 LOADER.GetMapImageN(20)->DrawFull(curPos);
 
-            // Wegbauzeug
+            // not building roads, no further action needed
             if(rb.mode == RM_DISABLED)
                 continue;
 
+            // we are in road build mode
+            // highlight current route pt
             if(rb.point == curPt)
+            {
                 LOADER.GetMapImageN(21)->DrawFull(curPos);
+                continue;
+            }
 
-            int altitude = GetWorld().GetNode(rb.point).altitude;
-
-            for(unsigned dir = 0; dir < Direction::COUNT; ++dir)
+            // ensure that curPt is a neighbour of rb.point
+            if(!helpers::contains(road_points, curPt))
```

Comment:
Currently this is the only usage of `road_points`. If you are not using indexOf as a cheaper way than GetNeighbour (also the latter is more clear) then you can remove the container completely and just put the loop here.

Reply:
after switching back to an array i am more comfortable to use road_points as "cache" for the backward direction lookup